src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Shapes/Basic_Shapes.xaml.cs

```diff
@@ -0,0 +1,651 @@
+				using (var output = await file.OpenAsync(FileAccessMode.ReadWrite))
+				{
+					var encoder = await BitmapEncoder.CreateAsync(BitmapEncoder.PngEncoderId, output);
+					encoder.SetSoftwareBitmap(SoftwareBitmap.CreateCopyFromBuffer(await renderer.GetPixelsAsync(), BitmapPixelFormat.Bgra8, renderer.PixelWidth, renderer.PixelHeight));
+					await encoder.FlushAsync();
+					await output.FlushAsync();
+				}
+			}
+
+			_root.Visibility = Visibility.Visible;
+			_testZone.Child = null;
+#endif
+		}
+
+#if __IOS__ // This is the base work for a fix in the UITests in order to highly increase Screenshot speed. It will be removed by https://github.com/unoplatform/Uno.UITest/issues/39
+		private static byte[] RenderAsPng(FrameworkElement elt)
+		{
+			UIImage img;
+			try
+			{
+				UIGraphics.BeginImageContextWithOptions(new Size(elt.ActualWidth, elt.ActualHeight), true, UIScreen.MainScreen.Scale);
+				var ctx = UIGraphics.GetCurrentContext();
+				ctx.SetFillColor(Colors.White);
+				elt.Layer.RenderInContext(ctx);
+				img = UIGraphics.GetImageFromCurrentImageContext();
+			}
+			finally
+			{
+				UIGraphics.EndImageContext();
+			}
+
+			using (img)
+			{
+				return img.AsPNG().ToArray();
+			}
+		}
+#else
+		private static byte[] RenderAsPng(FrameworkElement elt)
+			=> throw new NotImplementedException("Not supported yet on this platform");
+#endif
+
+		public string RunTests(string testNames)
+		{
+			TestResult = "";
+
+			var tests = testNames.Split(';', StringSplitOptions.RemoveEmptyEntries);
+			var id = Guid.NewGuid().ToString("N");
+
+			((DependencyObject)this).Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => RunTestsCore(tests));
+
+			return id;
+
+			async Task RunTestsCore(string[] strings)
+			{
+				var result = new StringBuilder();
+				result.AppendLine(id);
+
+				foreach (var test in strings)
+				{
+					result.Append(test);
+					result.Append(';');
+					try
+					{
+						var elt = await RenderById(test);
+						var testResult = RenderAsPng(elt);
+
+						result.Append("SUCCESS;");
+						result.Append(Convert.ToBase64String(testResult));
+					}
+					catch (Exception e)
+					{
+						result.Append("ERROR;");
+						result.Append(Convert.ToBase64String(Encoding.UTF8.GetBytes(e.ToString())));
+					}
+
+					result.AppendLine();
+				}
+
+				TestResult = result.ToString();
+			}
+		}
+
+		private void RenderById(object sender, RoutedEventArgs e)
+			//=> ((DependencyObject)this).Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => RenderById(_idInput.Text));
+			=> RunTests(_idInput.Text);
+
+		private async Task<FrameworkElement> RenderById(string id)
+		{
+			var tcs = new TaskCompletionSource<object>();
+
+			using var timeout = Debugger.IsAttached ? default : new CancellationTokenSource(TimeSpan.FromMilliseconds(1500));
+			using var reg = Debugger.IsAttached ? default : timeout.Token.Register(() => tcs.TrySetCanceled());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'reg'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5363624)